S01E02_BeSquare/SquareFactory.cs

```diff
@@ -15,6 +15,10 @@ public void AddMaterial(int width, int height)
             if (height <= 0)
                 throw new ArgumentOutOfRangeException(nameof(height), "Height of material must be positive.");
 
+            // Added a new condition here, so it's not possible to add material, that is smaller than the size of wanted squares.
```

Comment:
There is no such constraint (that material must be at least square sized) in the method comments of the interface definition, so this new condition causes the implementation to implement a slightly different contract than defined by the interface.

This causes the code to not be correct. This change seems unnecessary.

It is true that no contract is 100% accurate and sometimes restrictions are "obvious" and do not need to be documented. However, in this case it seems excessive - if you feed not enough material into a factory, the simplest thing is just to make no squares from it and carry on.